From stable-bounces@linux.kernel.org Fri Feb  1 15:23:18 2008
From: Jan Andersson <jan@gaisler.com>
Date: Fri,  1 Feb 2008 15:16:59 -0800
Subject: USB: fix usbtest halt check on big endian systems
To: linux-usb@vger.kernel.org
Cc: Jan Andersson <jan@gaisler.com>, Greg Kroah-Hartman <gregkh@suse.de>, stable <stable@kernel.org>
Message-ID: <1201907929-23721-18-git-send-email-gregkh@suse.de>


From: Jan Andersson <jan@gaisler.com>

usbtest did not swap the received status information when checking for
a non-zero value and failed to discover halted endpoints on big endian
systems.

Signed-off-by: Jan Andersson <jan@gaisler.com>
Acked-by: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/usbtest.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -1151,6 +1151,7 @@ static int verify_halted (int ep, struct
 		dbg ("ep %02x couldn't get halt status, %d", ep, retval);
 		return retval;
 	}
+	le16_to_cpus(&status);
 	if (status != 1) {
 		dbg ("ep %02x bogus status: %04x != 1", ep, status);
 		return -EINVAL;
